src/mono/wasm/runtime/driver.c

```diff
@@ -674,12 +674,21 @@ mono_wasm_assembly_get_entry_point (MonoAssembly *assembly)
 			return method;
 
 		MonoClass *klass = mono_method_get_class (method);
-		char *async_name = strdup (name);
+		char *async_name = malloc (name_length + 1);
+		sprintf (async_name, "%s$", name);
 
+		// look for "<Name>$"
+		MonoMethodSignature *sig = mono_method_get_signature (method, image, mono_method_get_token (method));
+		MonoMethod *async_method = mono_class_get_method_from_name (klass, async_name, mono_signature_get_param_count (sig));
+		if (async_method != NULL) {
+			free (async_name);
+			return async_method;
+		}
+
+		// look for "Name" by trimming the first and last character of "<Name>"
 		async_name [name_length - 1] = '\0';
```

Comment:
This will zero out the `$`, but you also need to zero out the `>`.

Reply:
Read it closer.

Reply:
Ah right, because you aren't null-terminating it and you end up with `<Name\0$`. I still think you include the null terminator—way too much of our code depends on that—and should zero out both characters.

Reply:
The buffer was definitely too short, switched to snprintf here (and in another place) to fix termination.  I don't see a need to null out the $ it is harmless.